pints/_transformation.py

```diff
@@ -0,0 +1,1120 @@
+        performed on :class:`LogPDF` and/or that requires ``evaluateS1()``;
+        e.g. not necessary if it's used for :class:`ErrorMeasure` without
+        :meth:`ErrorMeasure.evaluateS1()`.
+        """
+        return np.log(np.abs(np.linalg.det(self.jacobian(q))))
+
+    def log_jacobian_det_S1(self, q):
+        """
+        Computes the logarithm of the absolute value of the determinant of the
+        Jacobian, and returns the result plus the partial derivatives of the
+        result with respect to the parameters.
+
+        The returned data is a tuple ``(S, S')`` where ``S`` is a scalar value
+        and ``S'`` is a sequence of length ``n_parameters``.
+
+        Note that the derivative returned is of the log of the determinant of
+        the Jacobian, so ``S' = d/dq log(|det(J(q))|)``, evaluated at input.
+
+        *This is an optional method.* It is needed when transformation is
+        performed on :class:`LogPDF` and that requires ``evaluateS1()``.
+        """
+        # Directly calculate the derivative using jacobian_S1()
+        #
+        # d/dq log(|det(J(q))|) = 1/|det(J(q))| * sign(det(J(q)))
+        #                         * d/dq det(J(q))
+        #
+        # The second term is given by Eq. 46 in the Matrix Cookbook (2012)
+        # http://www.math.uwaterloo.ca/~hwolkowi/matrixcookbook.pdf
+        #
+        # d/dq det(J(q)) = det(J) Tr(J^{-1} d/dq J)
+        #
+        # Therefore
+        #
+        # d/dq log(|det(J(q))|) = Tr(J^{-1} d/dq J)
+        #
+        q = pints.vector(q)
+        jac, jac_S1 = self.jacobian_S1(q)
+        out_S1 = np.zeros(q.shape)
+        for i, jac_S1_i in enumerate(jac_S1):
+            out_S1[i] = np.trace(np.matmul(np.linalg.pinv(jac), jac_S1_i))
+        return self.log_jacobian_det(q), out_S1
+
+    def multiple_to_model(self, qs):
+        """
+        Transforms a series of parameter vectors ``qs`` from the search space
+        to the model space. ``qs`` must be provided in the shape
+        (``n_vectors``, ``n_parameters``).
+        """
+        qs = np.asarray(qs)
+        ps = np.zeros(qs.shape)
+        for i, q in enumerate(qs):
+            ps[i, :] = self.to_model(q)
+        return ps
+
+    def n_parameters(self):
+        """
+        Returns the dimension of the parameter space this transformation is
+        defined over.
+        """
+        raise NotImplementedError
+
+    def to_model(self, q):
+        """
+        Transforms a parameter vector ``q`` from the search space to the model
+        space.
+        """
+        raise NotImplementedError
+
+    def to_search(self, p):
+        """
+        Transforms a parameter vector ``p`` from the model space to the search
+        space.
+        """
+        raise NotImplementedError
+
+
+class ElementWiseTransformation(Transformation):
+    r"""
+    Abstract base class for objects that provide *element-wise* transformations
```

Comment:
Maybe mention in docs that this is a subclass of Transformation? (Not a separate thing)

Reply:
Done